jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt

```diff
@@ -11,13 +11,11 @@ class CodeWhispererLanguageManager {
     private val supportedLanguage = setOf(
         CodewhispererLanguage.Java.toString(),
         CodewhispererLanguage.Python.toString(),
-        CodewhispererLanguage.Javascript.toString()
+        CodewhispererLanguage.Javascript.toString(),
+        "jsx harmony"
     )
 
-    fun isLanguageSupported(language: ProgrammingLanguage): Boolean {
-        val mappedLanguage = getParentLanguage(language)
-        return supportedLanguage.contains(mappedLanguage.languageName)
-    }
+    fun isLanguageSupported(language: ProgrammingLanguage): Boolean = supportedLanguage.contains(language.languageName)
```

Comment:
instead of adding "jsx harmony" to the supported language set, can we do:
```
val mappedLanguage = getParentLanguage(language).toCodeWhispererLanguage()
return supportedLanguage.contains(mappedLanguage.toString())
```

Reply:
and in `getParentLanguage` map "jsx harmony" to javascript?

Reply:
yepp, updated with some tweak, we can't call `getParentLanguage()` here since it will map `jsx` -> `javascript`

Reply:
instead 
```
fun isLanguageSupported(language: ProgrammingLanguage): Boolean {
        val cwsprLanguage = language.toCodeWhispererLanguage()
        return cwsprLanguage != CodewhispererLanguage.Unknown && cwsprLanguage != CodewhispererLanguage.Plaintext
    }
```

Reply:
we check the transformed codewhispererLanguage type directly which have: java, python, javascript, jsx, plaintext, unknown

Reply:
sorry is there any downside for mapping `jsx` to `javascript` for checking support purpose?

Reply:
actually if we just check not `unkown` and not `plaintext`, we don't need the supported language set at all. But I think it's safer to still check against the parent